src/Rules/RuleTrait.php

```diff
@@ -0,0 +1,67 @@
+<?php
+
+declare(strict_types=1);
+
+namespace TwigCsFixer\Rules;
+
+use TwigCsFixer\Report\Report;
+use TwigCsFixer\Report\Violation;
+use TwigCsFixer\Report\ViolationId;
+
+trait RuleTrait
+{
+    private ?Report $report = null;
+
+    /**
+     * @var list<ViolationId>
+     */
+    private array $ignoredViolations = [];
+
+    public function getName(): string
+    {
+        return static::class;
+    }
+
+    public function getShortName(): string
+    {
+        $shortName = (new \ReflectionClass($this))->getShortName();
+
+        return str_ends_with($shortName, 'Rule') ? substr($shortName, 0, -4) : $shortName;
+    }
+
+    private function addMessage(
+        int $messageType,
+        string $message,
+        string $fileName,
+        ?int $line = null,
+        ?int $position = null,
+        ?string $messageId = null
+    ): bool {
+        $id = new ViolationId(
+            $this->getShortName(),
+            $messageId ?? ucfirst(strtolower(Violation::getLevelAsString($messageType))),
+            $line,
+            $position,
+        );
+        foreach ($this->ignoredViolations as $ignoredViolation) {
+            if ($ignoredViolation->match($id)) {
+                return false;
+            }
+        }
+
+        $report = $this->report;
```

Comment:
Instead of this, why not pass this as an argument to the method? Then the trait becomes disconnected from this property having to exist.

Reply:
Anyway I still need the check for the AbstractRule because the fixer will also use the method but without a report